### ISSUE TITLE
Using local files in RotationModel 

### ISSUE THREAD

**Issue by natbutter** [GPlates/gplately#261]
Running [02-PlateReconstruction Notebook](https://github.com/GPlates/gplately/blob/master/Notebooks/02-PlateReconstructions.ipynb) on the provided _local files_ results in the below error. Using the `PlateModelManager` web server files seems good.

Might be related to  #180 as is the same error but tested with older various versions of gplately with same result. I am working on Linux in a Docker container. Could be some inconsistency in my [environment.txt](https://github.com/user-attachments/files/16739005/environment.txt).

To reproduce, checkout the repo `git clone https://github.com/GPlates/gplately`

And run through the notebook
```
import gplately
from gplately import pygplates
import numpy as np
import glob, os
import matplotlib.pyplot as plt
import cartopy.crs as ccrs
import cartopy.mpl.gridliner as grd
from matplotlib.patches import Patch
from matplotlib.patches import FancyArrowPatch
from matplotlib.lines import Line2D
from plate_model_manager import PlateModelManager
```

Set the cell to `True` to use local files:
```
# set True to use Method 2
use_local_files = True
```

Load in the provided local files
```
if use_local_files:

    # Directory to plate model files
    input_directory = "./NotebookFiles/Muller_etal_2019_PlateMotionModel_v2.0_Tectonics/"

    # Locate rotation files and set up the RotationModel object
    rotation_filenames = glob.glob(os.path.join(input_directory, '*.rot'))
    rotation_model = pygplates.RotationModel(rotation_filenames)

    # Locate topology feature files and set up a FeatureCollection object 
    topology_filenames = glob.glob(os.path.join(input_directory, '*.gpml'))
    topology_features = pygplates.FeatureCollection()
    for topology_filename in topology_filenames:
        # (omit files with the string "inactive" in the filepath)
        if "Inactive" not in topology_filename:
            topology_features.add( pygplates.FeatureCollection(topology_filename) )
        else:
            topology_filenames.remove(topology_filename)

    # Locate static polygons and set up another FeatureCollection object
    static_polygon_file = input_directory+"StaticGeometries/StaticPolygons/Global_EarthByte_GPlates_PresentDay_StaticPlatePolygons_2019_v1.shp"
    static_polygons = pygplates.FeatureCollection(static_polygon_file)

```

Then fail when creating the `PlateReconstruction` object.
```
model = gplately.PlateReconstruction(rotation_model, topology_features, static_polygons)
```

> ---------------------------------------------------------------------------
> ArgumentError                             Traceback (most recent call last)
> Cell In[4], line 1
> ----> 1 model = gplately.PlateReconstruction(rotation_model, topology_features, static_polygons)
> 
> File /build/miniconda3/lib/python3.8/site-packages/gplately/reconstruction.py:55, in PlateReconstruction.__init__(self, rotation_model, topology_features, static_polygons, anchor_plate_id)
>      52     self.name = None
>      54 self.anchor_plate_id = int(anchor_plate_id)
> ---> 55 self.rotation_model = _RotationModel(
>      56     rotation_model, default_anchor_plate_id=anchor_plate_id
>      57 )
>      58 self.topology_features = _load_FeatureCollection(topology_features)
>      59 self.static_polygons = _load_FeatureCollection(static_polygons)
> 
> File /build/miniconda3/lib/python3.8/site-packages/gplately/pygplates.py:177, in RotationModel.__init__(self, rotation_features, default_anchor_plate_id)
>      67 def __init__(self, rotation_features, default_anchor_plate_id=0):
>      68     """**A RotationModel object can be constructed in three ways.**
>      69 
>      70     ---
>    (...)
>     175     ---
>     176     """
> --> 177     super(RotationModel, self).__init__(
>     178         rotation_features, default_anchor_plate_id=default_anchor_plate_id
>     179     )
>     180     self.filenames = []
>     182     # update filename list
> 
> ArgumentError: Python argument types in
>     RotationModel.__init__(RotationModel, RotationModel)
> did not match C++ signature:
>     __init__(boost::python::api::object, GPlatesUtils::non_null_intrusive_ptr<GPlatesApi::RotationModel, GPlatesUtils::NullIntrusivePointerHandler> rotation_model)
>     __init__(boost::python::api::object, GPlatesApi::FeatureCollectionSequenceFunctionArgument rotation_features, unsigned int reconstruction_tree_cache_size=150, bool extend_total_reconstruction_poles_to_distant_past=False, unsigned long default_anchor_plate_id=0)
>     __init__(boost::python::api::object, GPlatesApi::FeatureCollectionSequenceFunctionArgument rotation_features, unsigned int reconstruction_tree_cache_size=150, bool clone_rotation_features=True)

**Comment by natbutter** [comment]
Using the same data files with your provided docker file environment ([gplates/gplately](https://hub.docker.com/r/gplates/gplately/tags)) works fine. Most likely an issue with my environment.... somewhere.

**Comment by jcannon-gplates** [comment]
I think it was caused by using a version of pygplates < 0.29.

Soon we'll force the minimum pygplates version, which will give a different (more obvious) error message.

For reference: When the local `pygplates.RotationModel` is wrapped in another `pygplates.RotationModel` [here](https://github.com/GPlates/gplately/blob/e15e361b60f1b97645735d503df0dcdef4cd19d6/gplately/pygplates.py#L50) the `default_anchor_plate_id` parameter is only available for pygplates >= 0.29. The next gplately release will remove the wrapping of pygplates, and also include pygplates in the dependency list (since pygplates will soon support pip installs) and hence specify its minimum required version.
